support/modules-artifact-gen/single-file-artifact-gen

```diff
@@ -157,7 +157,7 @@ echo "$filename" > $filename_file
 
 # Create permissions file in plain text
 if [[ "$OSTYPE" == "darwin"* ]]; then
-  stat -f %A "${file}" > $permissions_file
+  stat -c %A "${file}" > $permissions_file
```

Comment:
This was long debugged by @mzedel and we decided on `-c %A` for a reason.

If I recall correctly, the issue was that "-c" did not behave the same way in Linux and in Mac OS. The [commit message](https://github.com/mendersoftware/mender/commit/60c7360d8bec6a9d863335898bcf915e0eadfdda) does not help much...

Reply:
> This was long debugged by @mzedel and we decided on `-c %A` for a reason.
> 
> If I recall correctly, the issue was that "-c" did not behave the same way in Linux and in Mac OS. The [commit message](https://github.com/mendersoftware/mender/commit/60c7360d8bec6a9d863335898bcf915e0eadfdda) does not help much...

I meant:
```bash
touch f; stat -c %A f;
-rw-r--r--
```

-f option does something else (--file-system display file system status instead of file status)
```bash
#touch f; stat -f %A f
stat: cannot read file system information for '%A': No such file or directory
  File: "f"
    ID: 100000400000019 Namelen: ?       Type: apfs
Block size: 4096       Fundamental block size: 4096
Blocks: Total: 29545461   Free: 2141099    Available: 1712250
Inodes: Total: 9223372036854775807 Free: 9223372036852788739
```

Reply:
it is just a typo nothing more

Reply:
It is not a typo, that I am sure about. You might be using a different version of stat than @mzedel. I wonder why this is not standard in different Mac OS.

The permissions file is expected to save the permissions in octal format, not in string. For example, in Linux:
```
touch f; stat -c %a f
644
```